feat: Add signal property for Request in `@sveltejs/adapter-node`

orisrightpaw opened the pull request:
Gives routes an AbortSignal ([as present in the Request object](https://developer.mozilla.org/en-US/docs/Web/API/Request/signal)) to know when the other end hangs up prematurely.  
  
Fixes #12823, but only for the Node adapter.  

---

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. Changesets that add features should be `minor` and those that fix bugs should be `patch`. Please prefix changeset messages with `feat:`, `fix:`, or `chore:`.

### Edits

- [x] Please ensure that 'Allow edits from maintainers' is checked. PRs without this option may be closed.

orisrightpaw commented:
Oops! This is a change to kit core, not the node adapter. 🤦

benmccann commented:
this would need some documentation. possibly in the `adapter-node` docs

Conduitry commented on .changeset/twelve-jokes-prove.md:
```diff
@@ -0,0 +1,6 @@
+---
+'@sveltejs/adapter-node': minor
```
It does touch `@sveltejs/kit/node`, though, and the Node adapter is bundled, so we need to re-build it and release a new version.

Conduitry replied:
On that note, `@sveltejs/kit/node`'s `getRequest` is also used in local development, in `preview` mode, and in production in the Vercel adapter. Is this going to have any impact on those?

orisrightpaw replied:
> On that note, `@sveltejs/kit/node`'s `getRequest` is also used in local development, in `preview` mode, and in production in the Vercel adapter. Is this going to have any impact on those?

Yes, it would. Apologies for not being clearer. Though, I wasn't aware that the vercel adapter used `getRequest`. I'll mark `@sveltejs/adapter-vercel` for a minor version, as well.

benmccann replied:
`adapter-vercel` does use it, but it's not bundled, so probably doesn't need to be listed here

orisrightpaw commented:
After a bit of testing behavior in Vercel, I couldn't get the request to hang up early without a body. It works fine on POST requests with data, but will ***not*** work on GET requests. I guess it's just a Vercel thing...? Someone else might be able to get it to work, but from what I can see, I don't think Vercel will tell you when the client cancels the request. I've removed the Vercel package from the changeset.    
  
On another note though, I squashed my commits and wrote some documentation. It's under the `adapter-node` docs, as @benmccann suggested.

benmccann commented on documentation/docs/25-build-and-deploy/40-adapter-node.md:
```diff
@@ -263,3 +263,49 @@ app.listen(3000, () => {
+
+## Aborted requests
+
+> [!NOTE] This feature is available in `vite dev` and `vite preview` servers.
+
```
I don't think we necessarily need to mention this here as basically all features are available during `dev` and `preview`
```suggestion
```

benmccann commented on documentation/docs/25-build-and-deploy/40-adapter-node.md:
```diff
@@ -263,3 +263,49 @@ app.listen(3000, () => {
+export async function GET({ request, params }) {
+    const stream = await s3.getObject("bucket", params.slug, {
+        signal: request.signal
+    }) 
```
```suggestion
    }) ;
```

benmccann commented on documentation/docs/25-build-and-deploy/40-adapter-node.md:
```diff
@@ -263,3 +263,49 @@ app.listen(3000, () => {
+    return new Response(stream, {
+        headers: {
+            "Content-Type": "text/event-stream",
+        },
```
```suggestion
        }
```

benmccann commented on documentation/docs/25-build-and-deploy/40-adapter-node.md:
```diff
@@ -263,3 +263,49 @@ app.listen(3000, () => {
+            request.signal.onabort = () => {
+                clearInterval(interval);
+            };
+        },
```
```suggestion
        }
```

abdelfattahradwan commented:
Any plans to merge this? Some apps I work on utilise SSE and would benefit from being able to detect when the request is aborted.
